beacon-chain/rpc/validator/status.go

```diff
@@ -88,6 +88,101 @@ func (vs *Server) MultipleValidatorStatus(
 	}, nil
 }
 
+// CheckDoppelGanger checks if the provided keys are currently active in the network.
+func (vs *Server) CheckDoppelGanger(ctx context.Context, req *ethpb.DoppelGangerRequest) (*ethpb.DoppelGangerResponse, error) {
+	if vs.SyncChecker.Syncing() {
+		return nil, status.Errorf(codes.Unavailable, "Syncing to latest head, not ready to respond")
+	}
+	if req == nil || req.ValidatorRequests == nil || len(req.ValidatorRequests) == 0 {
+		return &ethpb.DoppelGangerResponse{
+			Responses: []*ethpb.DoppelGangerResponse_ValidatorResponse{},
+		}, nil
+	}
+	headState, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Internal, "Could not get head state")
+	}
+	// We walk back from the current head state to the state at the beginning of the previous 2 epochs.
+	// Where S_i , i := 0,1,2. i = 0 would signify the current head state in this epoch.
+	currEpoch := helpers.SlotToEpoch(headState.Slot())
+	previousEpoch, err := currEpoch.SafeSub(1)
+	if err != nil {
+		previousEpoch = currEpoch
+	}
```

Comment:
Why would this error? 
Please log something and do not ignore this error.
```suggestion
	if err != nil {
		log.WithError(err).Error("Could not determine previous epoch")	
		previousEpoch = currEpoch
	}
```

Reply:
This is to prevent an overflow `( currEpoch <= 1)` , in this case the doppelganger check will be skipped anyway as it is at the start of the chain.